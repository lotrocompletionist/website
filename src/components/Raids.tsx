import * as React from "react";

export const Raids = () => <h2>Raids</h2>;
