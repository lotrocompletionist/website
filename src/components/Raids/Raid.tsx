import * as React from "react";
import {
  CardContent,
  Media,
  MediaLeft,
  MediaContent,
  Content,
  Image,
  Title,
  Subtitle
} from "bloomer";
import { LevelRange } from "@app/components/Shared/LevelRange/LevelRange";
import { Bosses } from "@app/components/Shared/Bosses/Bosses";
import { IRaid } from "@app/data/raid";

export interface RaidProps {
  raid: IRaid;
}

export function Raid(props: RaidProps) {
  return (
    <>
      <CardContent>
        <Media>
          <MediaLeft>
            <Image isSize="48x48" src="https://via.placeholder.com/96x96" />
          </MediaLeft>
          <MediaContent>
            <Title isSize={4}>{props.raid.name}</Title>
            <Subtitle isSize={6}>
              {props.raid.abbreviations.join(", ")}
            </Subtitle>
          </MediaContent>
        </Media>
        <Content>
          {props.raid.cluster}
          <br />
          <small>
            <LevelRange range={props.raid.level} />
            <br />
            <Bosses bosses={props.raid.bosses} />
          </small>
        </Content>
      </CardContent>
    </>
  );
}
