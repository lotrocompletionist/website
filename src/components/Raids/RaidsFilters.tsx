import * as React from "react";
import { clusters, regions } from "@app/data/raids";
import { DropdownFilter } from "@app/components/Shared/Filtering/DropdownFilter";
import { DropdownCheckboxFilterItem } from "@app/components/Shared/Filtering/DropdownFilterItem";

interface RaidsProps {
  onClusterChange: React.ReactEventHandler<HTMLInputElement>;
  onRegionChange: React.ReactEventHandler<HTMLInputElement>;
}

export function RaidsFilters(props: RaidsProps) {
  return (
    <>
      <DropdownFilter
        filterItem={DropdownCheckboxFilterItem}
        header="Cluster"
        values={clusters}
        onChange={props.onClusterChange}
      />
      <DropdownFilter
        filterItem={DropdownCheckboxFilterItem}
        header="Region"
        values={regions}
        onChange={props.onRegionChange}
      />
    </>
  );
}
