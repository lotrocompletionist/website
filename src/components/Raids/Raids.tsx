import * as React from "react";

import { Raid } from "./Raid";
import { raids, IRaid } from "@app/data/raid";
import { remove } from "lodash";
import { RaidsFilters } from "./RaidsFilters";

interface RaidsProps {}

interface RaidsState {
  filteredRaids: IRaid[];
  filteredClusters: string[];
  filteredRegions: string[];
}

export default class Raids extends React.Component<RaidsProps, RaidsState> {
  constructor(props: RaidsProps) {
    super(props);

    this.state = {
      filteredRaids: raids,
      filteredClusters: [],
      filteredRegions: []
    };
  }

  onClusterChange(event: React.ChangeEvent<HTMLInputElement>) {
    const updatedClusters = event.target.checked
      ? [...this.state.filteredClusters, event.target.value]
      : remove(this.state.filteredClusters, event.target.value);

    const filteredClusters = this.setState({
      filteredClusters: updatedClusters,
      filteredRaids: this.filterRaids(
        updatedClusters,
        this.state.filteredRegions
      )
    });
  }
  onRegionChange(event: React.ChangeEvent<HTMLInputElement>) {
    const updatedRegions = event.target.checked
      ? [...this.state.filteredRegions, event.target.value]
      : remove(this.state.filteredRegions, event.target.value);

    this.setState({
      filteredRegions: updatedRegions,
      filteredRaids: this.filterRaids(
        this.state.filteredClusters,
        updatedRegions
      )
    });
  }

  filterRaids(filteredClusters: string[], filteredRegions: string[]) {
    const matchesFilteredClusters = (worldInstance: IRaid) =>
      filteredClusters.length === 0 ||
      filteredClusters.indexOf(worldInstance.cluster) !== -1;

    const matchesFilteredRegions = (worldInstance: IRaid) =>
      filteredRegions.length === 0 ||
      filteredRegions.indexOf(worldInstance.region) !== -1;

    return raids.filter(
      raid => matchesFilteredClusters(raid) && matchesFilteredRegions(raid)
    );
  }

  render() {
    return (
      <>
        <RaidsFilters
          onClusterChange={this.onClusterChange.bind(this)}
          onRegionChange={this.onRegionChange.bind(this)}
        />
        {this.state.filteredRaids.map(raid => (
          <Raid key={raid.id} raid={raid} />
        ))}
      </>
    );
  }
}
