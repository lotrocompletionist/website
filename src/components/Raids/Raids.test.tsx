import React from "react";
import renderer from "react-test-renderer";
import { default as Raids } from "./Raids";

test("Raids renders correctly", () => {
  const tree = renderer.create(<Raids />).toJSON();
  expect(tree).toMatchSnapshot();
});
