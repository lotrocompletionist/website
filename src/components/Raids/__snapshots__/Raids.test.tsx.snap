// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Raids> renders correctly 1`] = `
Array [
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Helegrod: Drake Wing
        </h1>
        <h2
          className="subtitle is-6"
        >
          Drake-wing
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Shadows of Angmar
      <br />
      <small>
        Level: 
        50
         - 115
        <br />
        Bosses: 
        Grisgart, Praunk-Goj, Rémgoj, Burzdyr, Burzgost
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Helegrod: Spider Wing
        </h1>
        <h2
          className="subtitle is-6"
        >
          Spider-wing
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Shadows of Angmar
      <br />
      <small>
        Level: 
        50
         - 115
        <br />
        Bosses: 
        Adhargal, Ansach, Breosal, Grisgart, Zaudru
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Helegrod: Giant Wing
        </h1>
        <h2
          className="subtitle is-6"
        >
          Giant-wing
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Shadows of Angmar
      <br />
      <small>
        Level: 
        50
         - 115
        <br />
        Bosses: 
        Storvâgûn
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Helegrod: Dragon Wing
        </h1>
        <h2
          className="subtitle is-6"
        >
          Thorog
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Shadows of Angmar
      <br />
      <small>
        Level: 
        50
         - 115
        <br />
        Bosses: 
        Coldbear, Drugoth the Death-monger, Thorog
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          The Rift of Nûrz Ghâshu
        </h1>
        <h2
          className="subtitle is-6"
        >
          Rift
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Shadows of Angmar
      <br />
      <small>
        Level: 
        50
        
        <br />
        Bosses: 
        Barz, Zurm, Frûz, Zogtark, Narnûlubat, Shadow-eater, Thrâng, Thaurlach
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Bogbereth
        </h1>
        <h2
          className="subtitle is-6"
        >
          Bogbereth
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Shadows of Angmar
      <br />
      <small>
        Level: 
        50
        
        <br />
        Bosses: 
        Bogbereth
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Ferndúr the Virulent
        </h1>
        <h2
          className="subtitle is-6"
        >
          Ferndur
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Shadows of Angmar
      <br />
      <small>
        Level: 
        50
        
        <br />
        Bosses: 
        Ferndúr
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          The Vile Maw
        </h1>
        <h2
          className="subtitle is-6"
        >
          VM/Watcher
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Mines of Moria
      <br />
      <small>
        Level: 
        60
        
        <br />
        Bosses: 
        The Watcher in the Water
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Filikul
        </h1>
        <h2
          className="subtitle is-6"
        >
          Turtle
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Mines of Moria
      <br />
      <small>
        Level: 
        60
        
        <br />
        Bosses: 
        Nornúan
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Dâr Narbugud
        </h1>
        <h2
          className="subtitle is-6"
        >
          DN
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Scourge of Khazad-dûm
      <br />
      <small>
        Level: 
        60
        
        <br />
        Bosses: 
        Blagh, Rung, Zholuga, Flâgît, Îstum, The Blind One, The Mistress of Pestilence
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Barad Guldur
        </h1>
        <h2
          className="subtitle is-6"
        >
          BG
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Tower of Dol Guldur
      <br />
      <small>
        Level: 
        65
         - 115
        <br />
        Bosses: 
        Dúrchest, Cargaraf, Morgaraf, Lieutenant of Dol Guldur
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Ost Dunhoth
        </h1>
        <h2
          className="subtitle is-6"
        >
          OD
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      In Their Absence
      <br />
      <small>
        Level: 
        65
         - 115
        <br />
        Bosses: 
        Dúrchest, Dungarth, Vark, Kaveran, Durin's Bane, Frothmar, Baleleaf, Dourbark, Ivar, Matumáth, Matúrz-olog, Narslaug, Dúngorth, Gortheron
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Draigoch's Lair
        </h1>
        <h2
          className="subtitle is-6"
        >
          Draigoch
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Rise of Isengard
      <br />
      <small>
        Level: 
        75
        
        <br />
        Bosses: 
        Draigoch
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Tower of Orthanc
        </h1>
        <h2
          className="subtitle is-6"
        >
          ToO
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Rise of Isengard
      <br />
      <small>
        Level: 
        75
        
        <br />
        Bosses: 
        Crisiant, Usgarren, Kâlbak, Iorweth, Bukot, Saruman
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Flight to the Lonely Mountain
        </h1>
        <h2
          className="subtitle is-6"
        >
          Flight
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      The Road to Erebor
      <br />
      <small>
        Level: 
        20
         - 115
        <br />
        Bosses: 
        
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          The Fires of Smaug
        </h1>
        <h2
          className="subtitle is-6"
        >
          Smaug
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      The Road to Erebor
      <br />
      <small>
        Level: 
        20
         - 115
        <br />
        Bosses: 
        
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          The Battle for Erebor
        </h1>
        <h2
          className="subtitle is-6"
        >
          BfE
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      The Road to Erebor
      <br />
      <small>
        Level: 
        20
         - 115
        <br />
        Bosses: 
        
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          The Deeping Wall
        </h1>
        <h2
          className="subtitle is-6"
        >
          Wall, DW
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Helm's Deep
      <br />
      <small>
        Level: 
        10
         - 100
        <br />
        Bosses: 
        
      </small>
    </div>
  </div>,
  <div
    className="card-content"
  >
    <article
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            src="https://via.placeholder.com/96x96"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <h1
          className="title is-4"
        >
          Throne of the Dread Terror
        </h1>
        <h2
          className="subtitle is-6"
        >
          Throne
        </h2>
      </div>
    </article>
    <div
      className="content"
    >
      Battle of Pelennor
      <br />
      <small>
        Level: 
        115
        
        <br />
        Bosses: 
        Gothmog, Rakothas, Set-Akaji, Arpong-Dokh, Ku-Nerpag, Vadokhar, The Unbroken One, The Woe of Khand, The High Sorcerer of Harad, Gothmog
      </small>
    </div>
  </div>,
]
`;
